esp/services/ws_dfu/ws_dfuService.hpp

```diff
@@ -95,6 +95,27 @@ class CThorNodeGroupCache: public SuperHashTableOf<CThorNodeGroup, const char>
     CThorNodeGroup *lookup(const char* groupName, unsigned timeOutMinutes);
 };
 
+struct DFUReadAccessRequest
+{
+    const char *logicalName = nullptr;
+    const char *clusterName = nullptr;
+    IUserDescriptor *udesc  = nullptr;
+    CSecAccessType accessType;
+    unsigned expiry = 1;
+    bool refresh = false;
+    bool returnJsonTypeInfo = false;
+    bool returnBinTypeInfo = false;
+};
+
+struct DFUReadAccessResponse
+{
+    unsigned numParts = 0;
+    IArrayOf<IEspDFUPartLocations> dfuPartLocations;
+    IArrayOf<IEspDFUPartCopies> dfuPartCopies;
+    MemoryBuffer binLayout;
+    StringBuffer jsonLayout;
```

Comment:
@wangkx @rpastrana NB: removed expiryMinutes from response. *expiryTime* is in the encoded blob in the reply for dafilesrv's eyes only.